server/src/ga/GaWorker.ts

```diff
@@ -86,29 +86,39 @@ export class GaWorker {
 
   generateNewBots(startingPopulation: Bot[]): Bot[] {
     let generation = startingPopulation;
-    let fitnesses;
 
-    // Number of generations to iterate before returning to client
+    // Produce fitness scores from bots
+    let fitnesses = evaluate(
```

Comment:
Why do we get fitnesses here and then do the same thing later in the for loop? can this be cleaned?
Why not just move the fitnesses code to the top of the for loop.

Reply:
If `iterations=0` then we need to compute fitnesses outside the loop, otherwise a runtime error is thrown by `selectRandomWeightedNoReplacement` (technically iterations should never be 0 but I need that for certain experiments).
If the fitness code is at the top of the loop then when we call `selectRandomWeightedNoReplacement`, the fitnesses will be for the bots before they were mutated and invalid. Small change but thought I might as well make it.

Reply:
Ah good point... Hm.. why would we have iterations = 0? If its just for testing, I would be in favour of not commiting this into master because that case should never happen.

Reply:
Second point still holds though, even if `iterations != 1` the fitness values passed to `selectRandomWeightedNoReplacement` will be for the bots before they've mutated. Not sure if we can avoid two calls to `evaluate`.